### PR TITLE
Batch creator

### PR DESCRIPTION
**Pull request by MaxRodler**

### CODE REVIEW

File: src/include/BatchCreator.hpp [hunk 1]
```diff
@@ -11,22 +11,91 @@
...
+    /**
+     * @brief Initialisiert BatchCreator
+     * @details
+     * 
+     * @param filename
+     *
+     */
     BatchCreator(std::shared_ptr<parsing::FileData> fileData);
 
     std::string batchFileString;
```
Comment (DefinitelyNotSimon13) [inline src/include/BatchCreator.hpp:36]: why is the file string property still here?

File: src/sources/BatchCreator.cpp [hunk 2]
```diff
@@ -1,44 +1,84 @@
...
+ * @details Creates batch file from Arguments in JSON
  *
  * @copyright See LICENSE file
  */
 
 #include "BatchCreator.hpp"
+#include "LoggingWrapper.hpp"
 
-BatchCreator::BatchCreator(std::shared_ptr<parsing::FileData> fileData) {
+BatchCreator::BatchCreator(std::shared_ptr<parsing::FileData> fileData){
```
Comment (DefinitelyNotSimon13) [inline src/sources/BatchCreator.cpp:15]: Directly call createBatch upon instantiation

File: src/sources/BatchCreator.cpp [hunk 3]
```diff
@@ -1,44 +1,84 @@
...
 
 }
 
+void BatchCreator::createBatch() {
+    LOG_INFO << "Creating Batch file";
 
-
-void BatchCreator::writeBatch() {
+    this->batchFile.open(this->fileData->getOutputFile());
```
Comment (DefinitelyNotSimon13) [inline src/sources/BatchCreator.cpp:22]: check if file is actually open afterwards
file.good() or file.is_open()?

File: src/sources/BatchCreator.cpp [hunk 4]
```diff
@@ -1,44 +1,84 @@
...
+        this->batchFile << "/k \n";
     }
 }
 
-void BatchCreator::writeCommand() {
-    this->batchFileString = this->batchFileString + "\"";
+void BatchCreator::writeCommands() {
+    LOG_INFO << "writing Commands";
+    this->batchFile << "\"";
     for (const std::string command : this->fileData->getCommands()) {
```
Comment (DefinitelyNotSimon13) [inline src/sources/BatchCreator.cpp:47]: make std::string a reference

File: src/sources/BatchCreator.cpp [hunk 5]
```diff
@@ -1,44 +1,84 @@
...
-        this->batchFileString = this->batchFileString + command + "&& \n";
+        this->batchFile << command << "&& \n";
+    }
+}
+
+void BatchCreator::writeEnvVariables() {
+    LOG_INFO << "writing Environment Variables";
+    for (const std::tuple env : this->fileData->getEnvironmentVariables()) {
```
Comment (DefinitelyNotSimon13) [inline src/sources/BatchCreator.cpp:54]: Make tuple a reference

File: src/sources/BatchCreator.cpp [hunk 6]
```diff
@@ -1,44 +1,84 @@
...
+        this->batchFile << "set " << std::get<0>(env) << "=" 
+                        << std::get<1>(env) << " && \n";
+    }
+}
+
+void BatchCreator::writePathVariables() {
+    LOG_INFO << "writing Path Variables";
+    this->batchFile << "set path=";
+    for (const std::string path : this->fileData->getPathValues()) {
```
Comment (DefinitelyNotSimon13) [inline src/sources/BatchCreator.cpp:63]: make string a reference

File: src/sources/BatchCreator.cpp [hunk 7]
```diff
@@ -1,44 +1,84 @@
...
+        this->batchFile << path << ";\n";
     }
+    this->batchFile << "%path%";
+}
+
+void BatchCreator::writeApp() {
+    std::string appName = this->fileData->getOutputFile();
```
Comment (DefinitelyNotSimon13) [inline src/sources/BatchCreator.cpp:70]: what does this do?

**Review by DefinitelyNotSimon13 — CHANGES_REQUESTED** [review]

File: src/sources/BatchCreator.cpp (hunk 7 above)
Comment (MaxRodler) [reply src/sources/BatchCreator.cpp:70]: Holt den Namen der Anwendung damit dieser beim Staten mit ausgegeben werden kann.
(Wird in der nächsten Zeile formatiert: `appName = appName.substr(...)` )

File: src/sources/BatchCreator.cpp (hunk 5 above)
Comment (MaxRodler) [reply src/sources/BatchCreator.cpp:54]: Geht nicht...

File: src/include/BatchCreator.hpp [hunk 8]
```diff
@@ -11,22 +11,90 @@
...
+     * @brief Initialisiert BatchCreator
+     * @details
+     *
+     * @param filename
+     *
+     */
     BatchCreator(std::shared_ptr<parsing::FileData> fileData);
 
-    std::string batchFileString;
     std::shared_ptr<parsing::FileData> fileData;
```
Comment (DefinitelyNotSimon13) [inline src/include/BatchCreator.hpp:36]: Should not be public

File: src/include/BatchCreator.hpp [hunk 9]
```diff
@@ -11,22 +11,90 @@
...
-    void writeBatch();
+
+    std::ofstream batchFile;
```
Comment (DefinitelyNotSimon13) [inline src/include/BatchCreator.hpp:38]: Should not be public - should be pointer? Not sure how it works exactly

File: src/include/BatchCreator.hpp [hunk 10]
```diff
@@ -11,22 +11,90 @@
...
+    void createBatch();
```
Comment (DefinitelyNotSimon13) [inline src/include/BatchCreator.hpp:39]: Should be private, since called by Constructor

File: src/sources/BatchCreator.cpp [hunk 11]
```diff
@@ -1,44 +1,96 @@
...
+        this->writeStart();
+        this->writeHideShell();
+        this->writeCommands();
+        this->writeEnvVariables();
+        this->writePathVariables();
+        this->writeApp();
+        this->writeEnd();
+    } else {
+        throw exceptions::FailedToOpenFileException(this->fileData->getOutputFile());
+    }
```
Comment (DefinitelyNotSimon13) [inline src/sources/BatchCreator.cpp:36]: For less nested code do:
```cpp
if(!this->batchFile.is_open()) {
    throw ...
}
```
and then call functions

**Review by DefinitelyNotSimon13 — CHANGES_REQUESTED** [review]

**Review by DefinitelyNotSimon13 — APPROVED** [review]

**Pull request merged**